### PR TITLE
fixed gicp guess handling

### PR DESCRIPTION
**Pull request by prclibo**
Denote final transformation result Y=X*guess. Gicp should either 1) optimize X from identity by comparing guess*source and target or 2) optimize Y starting from X by comparing source and target. Current GICP code messes these two approaches. In `computeTransformation`, it uses `base_transformation_ = guess` as the start of the optimization but use `output = guess * source` as the source cloud.

This PR fixed it to approach 1).

### CODE REVIEW

**Comment by VictorLamoine** [comment]
I think this deserves a little bit of cleaning but otherwise the problem is solved right?

**Comment by prclibo** [comment]
@VictorLamoine I think so

**Comment by prclibo** [comment]
https://github.com/PointCloudLibrary/pcl/issues/1147
This might be related to the following observation

> I got the testcase passed when by setting transform.translation () = Eigen::Vector3f (0.1, 0.2, 0.3);, which is already a large shift compared to the model size. Larger translation like (20, -8, 1) leads to test failure, probably due to numerial failure, I guess.

File: registration/include/pcl/registration/impl/gicp.hpp [hunk 1]
```diff
@@ -455,17 +456,18 @@ pcl::GeneralizedIterativeClosestPoint<PointSource, PointTarget>::computeTransfor
...
   // final_transformation_.block<3,3> (0,0) = (transformation_.block<3,3> (0,0)) * (guess.block<3,3> (0,0));
   // final_transformation_.block <3, 1> (0, 3) = transformation_.block <3, 1> (0, 3) + guess.rightCols<1>.block <3, 1> (0, 3);
-  final_transformation_.topLeftCorner (3,3) = previous_transformation_.topLeftCorner (3,3) * guess.topLeftCorner (3,3);
-  final_transformation_(0,3) = previous_transformation_(0,3) + guess(0,3);
-  final_transformation_(1,3) = previous_transformation_(1,3) + guess(1,3);
-  final_transformation_(2,3) = previous_transformation_(2,3) + guess(2,3);
+  // final_transformation_.topLeftCorner (3,3) = previous_transformation_.topLeftCorner (3,3) * guess.topLeftCorner (3,3);
+  // final_transformation_(0,3) = previous_transformation_(0,3) + guess(0,3);
+  // final_transformation_(1,3) = previous_transformation_(1,3) + guess(1,3);
+  // final_transformation_(2,3) = previous_transformation_(2,3) + guess(2,3);
```
Comment (SergioRAgostinho) [inline registration/include/pcl/registration/impl/gicp.hpp]: Any particular reason for keeping these commented? Why not simply remove them.

**Comment by SergioRAgostinho** [comment]
Note: incorporated changes from #972

**Comment by SergioRAgostinho** [comment]
@prclibo Can you clean up the commit so that we can incorporate it in 1.8.1? Thank you

File: registration/include/pcl/registration/impl/gicp.hpp (hunk 1 above)
Comment (prclibo) [inline registration/include/pcl/registration/impl/gicp.hpp]: > for some reason the static equivalent methode raises an error

I am not sure what this means. Maybe a Eigen template problem at some versions, but anyway I am removing these comments.

**Comment by SergioRAgostinho** [comment]
Also, please squash your last 4 commits into a single one, basically only the commits in which you're the corresponding author.

**Comment by prclibo** [comment]
@SergioRAgostinho OK I am newbie to rebase and I think I messed it up=( Any suggestions?

**Comment by SergioRAgostinho** [comment]
Whoops! This never happened to me before. Here's some [options](http://stackoverflow.com/questions/134882/undoing-a-git-rebase) based on Google's first results.

**Comment by SergioRAgostinho** [comment]
Some other ideas:
- Add pcl official repo as a remote
- Pull the current HEAD it to a new branch named `clone` 
- [Cherry pick](http://stackoverflow.com/a/9339460) the original commits in this PR. The two original ones 50065756599d499178698edb71bde2c0e651a40a and 1d8c7f94a13464eb01f250aa9a09c2de657feb1b were from #972 and then there were 4 or 5 which I think you squashed into a357ea80bbefe1a5be6a016827a7f546266fbe13. 
- Force push this into your pcl master on github.

**Comment by VictorLamoine** [comment]
Backup your work first :)

**Comment by prclibo** [comment]
@SergioRAgostinho @VictorLamoine thanks for the help! It seems fixed and I hope I didn't miss anything.

**Pull request merged**